os_fdp_adapters/xml-adapter/xml_to_fdp.py

```diff
@@ -0,0 +1,100 @@
+import sys
+import csv
+import json
+import tempfile
+import os
+import six
+
+sys.path.append('..')
+from common.fetchers import fetch_local_filename
+from common.urls import wrap
+
+csvDelimiter = ','
+
+
+def to_bytes(x):
+    if six.PY2 and type(x) is six.text_type:
+        return x.encode('utf8')
+    return x
+
+
+def gen_csv_file(url):
+    """
+    :param url: url can be the absolute address of an xml file, or  an xml file located at web (http://..../<name>.xml
+    :return: list of csv files, each contains a data-table extracted from the input xml file
+    """
+    orginBaseName = os.path.basename(url).strip(' ')
+    if url.find('http') == -1:
+        url = os.path.abspath(url)
+    with tempfile.NamedTemporaryFile(delete=False) as fd:
+        cmd = "xml2csv {} -a 135 >> {}".format(url, fd.name)
+        os.system(cmd)
+        return split_into_single_csvs(fd.name, orginBaseName)
+
+
+def split_into_single_csvs(filename, orginBaseName, tableSeparator="\n\n\n"):
+    """
+    :param  filename: a file containing several data-tables, which are separated by blank lines
+    :return: a list of csv file names, each is csv file containing one data-table
+    """
+    with open(filename, 'r') as fd:
+        data = fd.read().split("file exist\n")[-1]
+        count = -1
+        csvLst = []
+        for oneTable in data.split(tableSeparator)[:-1]:
+            if oneTable != '':
+                count += 1
+                #tmpFile = ""
+                #with tempfile.NamedTemporaryFile(delete=False) as csvFd:
+                #    csvFd.write(str.encode(oneTable, 'utf8'))
+                #    tmpFile = csvFd.name
+                #path, base = os.path.split(tmpFile)
+                #csvFileName = os.path.join(path, orginBaseName+"_"+str(count)+".csv")
+                #os.renames(tmpFile , csvFileName)
+                testPath = "xml-adapter/tests"
+                baseName = os.path.splitext(orginBaseName)[0]
+                if count == 0:
+                    csvFileName = os.path.join(testPath, baseName + "#" + str(count) + ".xml.csv")
+                    with open(csvFileName, 'bw+') as csvFd:
+                        oneTable = '\r\n'.join(oneTable.split('\n')[2:])
+                        csvFd.write(str.encode(oneTable, 'utf8'))
+                else:
+                    csvFileName = os.path.join(testPath, baseName+"#"+str(count)+".xml.csv")
+                    with open(csvFileName, 'bw+') as csvFd:
+                        oneTable = '\r\n'.join(oneTable.split('\n'))
+                        csvFd.write(str.encode(oneTable, 'utf8'))
+                csvLst.append(csvFileName)
+    return csvLst
+
+
+filename = sys.argv[1]
+
+if filename.endswith('.csv'):
+    filename = fetch_local_filename(filename)
```

Comment:
This entire code block doesn't make sense.

`fetch_local_filename` downloads an external URL - but the filename here is not a valid URL (as you appended `.csv` to it).
Then you open the CSV file, read it, parse it and encode it again as CSV - why??